### PR TITLE
Display extra pages in goal info commands

### PR DESCRIPTION
**Pull request by jasonpaulos**
## Summary

With this PR, the commands `goal app info`, `goal account info`, and `goal account list --info` now display the number of extra pages an application has.

For example, the following were executed on a private network where apps 1, 2, 3, 4 have 1, 2, 3, 0 extra pages, respectively.
```
$ goal app info --app-id 3 -d DATA_DIR
Application ID:        3
Creator:               U4SKWGT6KYRA3DCMVPJLMMMNCP35UOYBFMQLEHLYDHO4C4N7GU7LLNQUHU
Approval hash:         Q7ODX32ILHMGO6KPSELVTO7RCVOUZ37RWYHISAWER7GVARJGKTWEWD37AM
Clear hash:            P7GEWDXXW5IONRW6XRIRVPJCT2XXEQGOBGG65VJPBUOYZEJCBZWTPHS3VQ
Extra program pages:   3
Max global byteslices: 2
Max global integers:   1
Max local byteslices:  4
Max local integers:    3
```

```
$ goal account info -a U4SKWGT6KYRA3DCMVPJLMMMNCP35UOYBFMQLEHLYDHO4C4N7GU7LLNQUHU -d DATA_DIR
Created Assets:
        <none>
Held Assets:
        <none>
Created Apps:
        ID 1, 1 extra page, global state used 0/1 uints, 0/2 byte slices
        ID 2, 2 extra pages, global state used 0/1 uints, 0/2 byte slices
        ID 3, 3 extra pages, global state used 0/1 uints, 0/2 byte slices
        ID 4, global state used 0/1 uints, 0/2 byte slices
Opted In Apps:
        <none>
```

```
$ goal account list --info -d DATA_DIR
[offline]       EHF6QYU3HCUUXIZAMPJ5RZWNDT37Y72NHVZ4XHWMYILPXWVU7G2G4PX32I   EHF6QYU3HCUUXIZAMPJ5RZWNDT37Y72NHVZ4XHWMYILPXWVU7G2G4PX32I   4000024000000000 microAlgos
Created Assets:
...
[offline]       U4SKWGT6KYRA3DCMVPJLMMMNCP35UOYBFMQLEHLYDHO4C4N7GU7LLNQUHU   U4SKWGT6KYRA3DCMVPJLMMMNCP35UOYBFMQLEHLYDHO4C4N7GU7LLNQUHU   1000006000004997 microAlgos     [created app IDs: 1, 2, 3, 4]
Created Assets:
        <none>
Held Assets:
        <none>
Created Apps:
        ID 1, 1 extra page, global state used 0/1 uints, 0/2 byte slices
        ID 2, 2 extra pages, global state used 0/1 uints, 0/2 byte slices
        ID 3, 3 extra pages, global state used 0/1 uints, 0/2 byte slices
        ID 4, global state used 0/1 uints, 0/2 byte slices
Opted In Apps:
        <none>
```

## Test Plan

Modified existing e2e test and expect test. (I know expect tests aren't currently running, but it passed locally.)

### CODE REVIEW

File: test/scripts/e2e_subs/e2e-app-extra-pages.sh [hunk 1]
```diff
@@ -97,8 +94,13 @@ fi
 
 RES=$(${gcmd} app info --app-id ${APPID}  2>&1 || true)
 PROGHASH="Approval hash:         7356635AKR4FJOOKXXBWNN6HDJ5U3O2YWAOSK6NZBPMOGIQSWCL2N74VT4"
+EXTRAPAGES="Extra program pages:   1"
 if [[ $RES != *"${PROGHASH}"* ]]; then
-    date '+app-extra-pages-test FAIL the application info should succeed %Y%m%d_%H%M%S'
+    date '+app-extra-pages-test FAIL the application approval program hash is incorrect %Y%m%d_%H%M%S'
```
Comment (jannotti) [inline test/scripts/e2e_subs/e2e-app-extra-pages.sh:99]: Might be nice to migrate toward doing this at the top of e2e .sh programs
```
filename=$(basename "$0")
scriptname="${filename%.*}"
date "+${scriptname} start %Y%m%d_%H%M%S"
```
So you can use $scriptname here (and elsewhere). No big deal, just makes copying testing for modification easier.

File: cmd/goal/account.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ import (
 	"github.com/algorand/go-algorand/config"
 	"github.com/algorand/go-algorand/crypto"
 	"github.com/algorand/go-algorand/crypto/passphrase"
-	v1 "github.com/algorand/go-algorand/daemon/algod/api/spec/v1"
+	generatedV2 "github.com/algorand/go-algorand/daemon/algod/api/server/v2/generated"
```
Comment (jannotti) [inline cmd/goal/account.go:34]: Can you summarize what we're doing with versioning that forces changes in this PR?  We don't have to increment the version for changes like this, right?  Maybe you're just migrating to the new version that we previously had to do?

**Review by jannotti — APPROVED** [review]
It seems good to me, assuming I've properly understood why there's so much churn around unrelated things (because of upgrade to v2, I'm assuming).

File: cmd/goal/account.go (hunk 2 above)
Comment (jasonpaulos) [reply cmd/goal/account.go:34]: Actually I had to upgrade to using the v2 endpoint because v1 does not return extra pages values for accounts.

File: cmd/goal/account.go [hunk 3]
```diff
@@ -563,103 +577,116 @@ func printAccountInfo(client libgoal.Client, address string, account v1.Account)
...
-	for _, id := range createdAssets {
-		assetParams := account.AssetParams[id]
-
-		name := assetParams.AssetName
-		if len(name) == 0 {
-			name = "<unnamed>"
+	for _, createdAsset := range createdAssets {
+		name := "<unnamed>"
+		if createdAsset.Params.Name != nil {
+			_, name = unicodePrintable(*createdAsset.Params.Name)
```
Comment (jasonpaulos) [inline cmd/goal/account.go:583]: It may be unnecessary to pipe the string fields through `unicodePrintable` after #2555, but I've left them in just in case.
